fdbclient/NativeAPI.actor.cpp

```diff
@@ -1578,6 +1578,12 @@ DatabaseContext::DatabaseContext(Reference<AsyncVar<Reference<IClusterConnection
 	smoothMidShardSize.reset(CLIENT_KNOBS->INIT_MID_SHARD_BYTES);
 	globalConfig = std::make_unique<GlobalConfig>(this);
 
+	if (g_network->isSimulated()) {
```

Comment:
Do you still need this even with the change in Knobs.cpp?

Reply:
At least with the current implementation, both need to be true to trace the transaction. Here's the relevant snippet from `generateSpanID`:

```
	if (transactionTracingSample) {
		return SpanContext(deterministicRandom()->randomUniqueID(),
		                   deterministicRandom()->randomUInt64(),
		                   deterministicRandom()->random01() <= FLOW_KNOBS->TRACING_SAMPLE_RATE
		                       ? TraceFlags::sampled
		                       : TraceFlags::unsampled);
        }
```

I think the idea here is to allow tracing control on an individual transaction level, but then also have a knob to control trace percentage overall. It's a little confusing having both of these options though.

Reply:
transactionTracingSample is set here based on getCommittedVersion: https://github.com/apple/foundationdb/blob/main/fdbclient/NativeAPI.actor.cpp#L6714

maybe if you set TRACING_SAMPLE_RATE to 1.0 (or 0.5) it would be sufficient?

Reply:
You're right, I didn't need this part :)

> maybe if you set TRACING_SAMPLE_RATE to 1.0 (or 0.5) it would be sufficient?

Do you mean instead of setting it to 0.01 under buggification? My thinking here was we want to test the transaction path with tracing in simulation, but not necessarily all the time. But if we are going to enable tracing in production, maybe we do want to test that path more often in simulation. Do you think I should up the 0.01 value to 0.5? (or 1.0)